code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -169,6 +170,85 @@
 	if(.)
 		name = "\improper Obrez Moderna" // wear it loud and proud
 
+/obj/item/gun/ballistic/rifle/rebarxbow
+	name = "Heated Rebar Crossbow"
+	desc = "Made from an inducer, iron rods, and some wire, this crossbow fires sharpened iron rods, made from the plentiful iron rods found stationwide. \
+			Requires a reload every shot - you can use a wrench on crossbar to try and force a second rod in, but risks a misfire, or worse..."
+	icon = 'icons/obj/weapons/guns/ballistic.dmi'
+	icon_state = "rebarxbow"
+	inhand_icon_state = "rebarxbow"
+	worn_icon_state = "rebarxbow"
+	rack_sound = 'sound/weapons/gun/sniper/rack.ogg'
+	must_hold_to_load = FALSE
+	mag_display = FALSE
+	empty_indicator = TRUE
+	bolt_type = BOLT_TYPE_LOCKING
+	semi_auto = FALSE
+	internal_magazine = TRUE
+	can_modify_ammo = TRUE
+	initial_caliber = CALIBER_REBAR
+	slot_flags = ITEM_SLOT_BACK|ITEM_SLOT_SUITSTORE
+	alternative_caliber = CALIBER_REBAR_FORCED
+	alternative_ammo_misfires = TRUE
+	var/draw_time = 3
+	bolt_wording = "bowstring"
+	magazine_wording = "rod"
+	cartridge_wording = "rod"
+	misfire_probability = 25
+	accepted_magazine_type = /obj/item/ammo_box/magazine/internal/boltaction/rebarxbow/normal
+	fire_sound = 'sound/items/syringeproj.ogg'
+	can_be_sawn_off = FALSE
+	tac_reloads = FALSE
+	SET_BASE_PIXEL(0, 0)
+
+/obj/item/gun/ballistic/rifle/rebarxbow/rack(mob/user = null)
+	if (bolt_locked)
+		drop_bolt(user)
+		return
+	balloon_alert(user, "bowstring loosened")
+	playsound(src, rack_sound, rack_sound_volume, rack_sound_vary)
+	process_chamber(FALSE, FALSE, FALSE)
+	bolt_locked = TRUE
+	update_appearance()
+
+/obj/item/gun/ballistic/rifle/rebarxbow/drop_bolt(mob/user = null)
+	if(!do_after(user, draw_time SECONDS, target = src))
+		return
+		playsound(src, bolt_drop_sound, bolt_drop_sound_volume, FALSE)
```

Comment:
Unindent following the return.

Reply:
You shouldn't have resolved this because you didn't do it. This code doesn't run.

Reply:
Sorry, been working on it on my end, didnt know a resolve went through before I did a commit.